rdmo/projects/progress.py

```diff
@@ -47,6 +47,48 @@ def compute_sets(values):
     return sets
 
 
+def compute_next_relevant_page(current_page, direction, catalog, resolved_conditions):
+    # compute the next relevant page based on resolved conditions.
+    max_iterations = len(catalog.pages)
+    iterations = 0
+
+    while iterations < max_iterations:
```

Comment:
This could be a for loop over catalog pages (which should be ordered) beginning with `current_page`. Maybe this would be more elegant, idk, you decide.

Reply:
Or a recursive function :smiling_imp:

Reply:
ah yes, a recursive function also goes there, it makes the function very compact and simple actually